src/tools/devtools/web-performance.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Debugging performance for web apps
+description: Learn how to use Chrome DevTools to debug web performance issues.
+---
+
+{{site.alert.note}}
+  The following requires Flutter version 3.14 or later.
+{{site.alert.end}}
+
+The Flutter framework emits timeline events as it works to build frames,
+draw scenes, and track other activity such as garbage collections.
+These events are exposed in the 
+[Chrome DevTools performance panel][] for debugging.
+
+You can also emit your own timeline events using the `dart:developer`
+[Timeline][] and [TimelineTask][] APIs for further performance analysis.
+
+[Chrome DevTools performance panel]: https://developer.chrome.com/docs/devtools/performance
+[Timeline]: https://api.flutter.dev/flutter/dart-developer/Timeline-class.html
+[TimelineTask]: https://api.flutter.dev/flutter/dart-developer/TimelineTask-class.html
+
+![Screenshot of the Chrome DevTools performance panel]({{site.url}}/assets/images/docs/tools/devtools/chrome-devtools-performance-panel.png)
+
+## Optional flags to enhance tracing
+
+- [debugProfileBuildsEnabled][]: Adds Timeline events for every Widget built.
+- [debugProfileBuildsEnabledUserWidgets][]: Adds Timeline events for every user-created Widget built.
+- [debugProfileLayoutsEnabled][]: Adds Timeline events for every RenderObject layout.
+- [debugProfilePaintsEnabled][]: Adds Timeline events for every RenderObject painted.
```

Comment:
Since these are referencing code types, they should generally be in code syntax:

```suggestion
- [debugProfileBuildsEnabled][]: Adds `Timeline` events for every `Widget` built.
- [debugProfileBuildsEnabledUserWidgets][]: Adds `Timeline` events for every user-created `Widget` built.
- [debugProfileLayoutsEnabled][]: Adds `Timeline` events for every `RenderObject` layout.
- [debugProfilePaintsEnabled][]: Adds `Timeline` events for every `RenderObject` painted.
```